src/sentry/incidents/endpoints/organization_alert_rule_details.py

```diff
@@ -93,46 +99,10 @@ def put(self, request: Request, organization, alert_rule) -> Response:
         )
 
         if serializer.is_valid():
-            if not self._verify_user_has_permission(request, alert_rule):
```

Comment:
Discussed removing this in [Slack](https://sentry.slack.com/archives/C058J6Y4HQU/p1690580097025369) - short version is this isn't actually used anywhere so we're not going to enforce it.